CMakeModules/FindMPI.cmake

```diff
@@ -1005,19 +1030,29 @@ foreach (lang C CXX Fortran)
           set(try_libs TRUE)
         endif()
     
-        #messagev("Looking for MPI_${lang}_COMPILER with names ${_MPI_${lang}_COMPILER_NAMES} at ${_MPI_PREFIX_PATH} + environment PATH")
+        messagev("Looking for MPI_${lang}_COMPILER with names ${_MPI_${lang}_COMPILER_NAMES} at ${_MPI_PREFIX_PATH}")
+        # Try and find the MPI compiler. First look in the hints directories
         find_program(MPI_${lang}_COMPILER
           NAMES  ${_MPI_${lang}_COMPILER_NAMES}
           HINTS  ${_MPI_PREFIX_PATH}
           PATH_SUFFIXES ${_BIN_SUFFIX}
-          ${PATHOPT})
+          NO_DEFAULT_PATH)
         interrogate_mpi_compiler(${lang} ${try_libs})
-        mark_as_advanced(MPI_${lang}_COMPILER)
+	if(NOT $MPI_${lang}_FOUND AND NOT PATHOPT STREQUAL NO_DEFAULT_PATH)
+	  # If the MPI compiler is not found then look in wider paths.
+          messagev("Looking for MPI_${lang}_COMPILER with names ${_MPI_${lang}_COMPILER_NAMES} at ${_MPI_PREFIX_PATH} + environment PATH")
+          find_program(MPI_${lang}_COMPILER
+            NAMES  ${_MPI_${lang}_COMPILER_NAMES}
+            HINTS  ${_MPI_PREFIX_PATH}
+            PATH_SUFFIXES ${_BIN_SUFFIX}
+            ${PATHOPT})
+          interrogate_mpi_compiler(${lang} ${try_libs})
+	endif()
     
         # last ditch try -- if nothing works so far, just try running the regular compiler and
         # see if we can create an MPI executable.
         set(regular_compiler_worked 0)
-        if (NOT MPI_${lang}_LIBRARIES OR NOT MPI_${lang}_INCLUDE_PATH)
+        if (NOT MPI_${lang}_FOUND)
```

Comment:
Yes, we are trying to find an MPI compiler. This is done via an interrogate_mpi_compiler call which sets the MPI_${lang}_FOUND variable based directly on the LIBRARIES and INCLUDE_PATH variables as above.